kubeshop/testkube
# ci: add SBOM and cosign for the binary and docker images

## Pull request description 
- Added creation of SBOMs files for all generated artefacts;
- added keyless signing of all artifacts with `cosign`;
- added keyless signing of the Docker images with `cosign`.

To verify the binary using `cosign`:
```
cosign verify-blob \
  --certificate-identity 'https://github.com/kubeshop/testkube/.github/workflows/release-dev.yaml@refs/tags/v1.11.24-beta001' \
  --certificate-oidc-issuer 'https://token.actions.githubusercontent.com' \
  --cert https://github.com/kubeshop/testkube/releases/download/v1.11.24-beta001/checksums.txt.pem \
  --signature https://github.com/kubeshop/testkube/releases/download/v1.11.24-beta001/checksums.txt.sig \
  ./checksums.txt
```

To verify the image using `cosign`:
```
cosign verify kubeshop/testkube-cypress-executor:1.11.24-beta007 --certificate-identity 'https://github.com/kubeshop/testkube/.github/workflows/docker-build-api-executors-beta-tag.yaml@refs/tags/v1.11.24-beta007' --certificate-oidc-issuer 'https://token.actions.githubusercontent.com'
```

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested on cluster

## Breaking changes

-

## Changes

-

## Fixes

-
